### DISCUSSION TITLE
The logic of include and exclude when loading resources with a Maven plugin.

### DISCUSSION THREAD

**Discussion by abing22333** [TongchengOpenSource/smart-doc#584] [General]
## Current Resource Loading Logic
1. Recursively resolve all dependencies first.
2. If exclude is present, apply the exclude logic; successfully matched dependencies are discarded.
3. If include is present, apply the include pattern; unmatched dependencies are discarded.
4. Load the remaining dependencies.

## Exclude Does Not Follow Maven's Exclusion Logic

In the case of spring-boot-starter-data-mongodb -> spring-data-mongodb,

in Maven, excluding `spring-boot-starter-data-mongodb` would remove all its dependencies.
However, in smart-doc, excluding `spring-boot-starter-data-mongodb` does not exclude `spring-data-mongodb`.

## Personal View
I believe that smart-doc should extend Maven's logic rather than implementing its own.

**Comment by shalousun** [comment]
The current dependencies should also be retrieved recursively; however, it would be beneficial to conduct detailed testing, as there may be discrepancies from what is expected.
